internal/rbd/controllerserver.go

```diff
@@ -833,6 +878,48 @@ func (cs *ControllerServer) DeleteVolume(
 	}
 	defer cs.OperationLocks.ReleaseDeleteLock(volumeID)
 
+	if isMigrationVolHandle(volumeID) {
+		log.DebugLog(ctx, "migration volume handle : %s", volumeID)
+		pMH, pErr := parseMigrationVolHandle(volumeID)
+		if pErr != nil {
+			log.DebugLog(ctx, "failed to parse migration volume handle: %s , err: %v", volumeID, pErr)
+
+			return nil, pErr
+		}
+		rv := &rbdVolume{}
+		if pMH != nil {
+			rv.RbdImageName = pMH.ImageName
+			rv.Monitors = pMH.Mons
+			rv.Pool = pMH.Pool
+		}
+		log.DebugLog(ctx, "filled rbd volume: %+v", rv)
+		monsMap := make(map[string]string)
+		monsMap["monitors"] = rv.Monitors
+		clusterID, cErr := getClusterIDFromMigrationVolume(monsMap)
+		if err != nil {
+			log.DebugLog(ctx, "failed to get ClusterID for the rbd image : %s", rv.RbdImageName)
+
+			return nil, cErr
+		}
+		rv.ClusterID = clusterID
+		err = rv.Connect(cr)
+		if err != nil {
+			log.DebugLog(ctx, "failed to get connected to the rbd image : %s", rv.RbdImageName)
+
+			return nil, err
+		}
+		defer rv.Destroy()
+		// if connected , delete it
+		err = deleteImage(ctx, rv, cr)
```

Comment:
check image exists before deleting need to handle not found error

Reply:
`deleteImage` try to get image and return error, so its handled already?

Reply:
deleteImage internally calls getImageID first and its returns the `ErrImageNotFound` error as we don't have a check for that here it always returns an internal server error.